app/uk/gov/hmrc/customsexportsinternalfrontend/controllers/actions/AuthenticatedAction.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customsexportsinternalfrontend.controllers.actions
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import play.api.mvc.Results.Forbidden
+import play.api.mvc._
+import play.api.{Configuration, Environment, Logger}
+import uk.gov.hmrc.auth.core.AuthProvider.PrivilegedApplication
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.{AuthProviders, AuthorisationException, AuthorisedFunctions, NoActiveSession}
+import uk.gov.hmrc.customsexportsinternalfrontend.config.AppConfig
+import uk.gov.hmrc.customsexportsinternalfrontend.connectors.StrideAuthConnector
+import uk.gov.hmrc.customsexportsinternalfrontend.controllers.exchanges.{AuthenticatedRequest, Operator}
+import uk.gov.hmrc.customsexportsinternalfrontend.views.html.unauthorized
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.play.bootstrap.config.AuthRedirects
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AuthenticatedAction @Inject()(
+  appConfig: AppConfig,
+  override val config: Configuration,
+  override val env: Environment,
+  override val authConnector: StrideAuthConnector,
+  mcc: MessagesControllerComponents,
+  unauthorizedPage: unauthorized
+) extends ActionBuilder[AuthenticatedRequest, AnyContent] with AuthorisedFunctions with AuthRedirects {
+
+  override implicit val executionContext: ExecutionContext = mcc.executionContext
+  override val parser: BodyParser[AnyContent] = mcc.parsers.defaultBodyParser
```

Comment:
Please Inject parser

Reply:
I see it now, I added this somewhere, but don't remember where

Reply:
Controllers all inject it if they need it